src/LineageTree/lineageTree.py

```diff
@@ -48,12 +45,198 @@ def get_next_id(self):
+
+    def fuse_lineage_tree(
+        self,
+        l1_root: int,
+        l2_root: int,
+        length_l1: int = 0,
+        length_l2: int = 0,
+        length: int = 1,
+    ):
+        """Fuses 2 lineages from the lineagetree object. The 2 lineages that are to be fused can have a longer
+        first node and the node of the resulting lineage can also be longer.
+
+        Args:
+            l1_root (int): Id of the first root
+            l2_root (int): Id of the second root
+            length_l1 (int, optional): The length of the branch that will be added on top of the first lineage. Defaults to 1.
+            length_l2 (int, optional): The length of the branch that will be added on top of the second lineage. Defaults to 1.
+            length (int, optional): The length of the branch that will be added on top of the resulting lineage. Defaults to 1.
+
+        Returns:
+            int: The id of the root of the new lineage.
+        """
+        if self.predecessor.get(l1_root) or self.predecessor.get(l2_root):
+            raise ValueError("Please select 2 roots.")
+        if self.time[l1_root] != self.time[l2_root]:
+            warnings.warn(
+                "Using lineagetrees that do not exist in the same timepoint. The operation will continue"
+            )
+        new_root1 = self.add_branch(l1_root, length_l1)
+        new_root2 = self.add_branch(l2_root, length_l2)
+        next_root1 = self[new_root1][0]
+        self.remove_nodes(new_root1)
+        self.successor[new_root2].append(next_root1)
+        self.predecessor[next_root1] = [new_root2]
+        new_branch = self.add_branch(new_root2, length)
+        self.labels[new_branch] = f"Fusion of {new_root1} and {new_root2}"
+        return new_branch
+
+    def copy_lineage(self, root):
+        """Copies a subtree and assigns it to new nodes.
+            Warning does not take into account the predecessor of the root node.
+
+        Args:
+            root (int): The root of the tree to be copied
+
+        Returns:
+            int: The root of the new tree.
+        """
+        new_nodes = {
+            old_node: self.get_next_id()
+            for old_node in self.get_sub_tree(root)
+        }
+        self.nodes.update(set(new_nodes.values()))
+        for old_node, new_node in new_nodes.items():
+            self.time[new_node] = self.time[old_node]
+            succ = self.successor.get(old_node, [])
+            if succ:
+                self.successor[new_node] = [new_nodes[n] for n in succ]
+            pred = self.predecessor.get(old_node, [])
+            if pred:
+                self.predecessor[new_node] = [new_nodes[n] for n in pred]
+            self.pos[new_node] = self.pos[old_node] + 0.5
+            self.time_nodes[self.time[old_node]].add(new_nodes[old_node])
+        new_root = new_nodes[root]
+        self.labels[new_root] = f"Copy of {root}"
+        if self.time[new_root] == 0:
+            self.roots.add(new_root)
```

Comment:
Are the roots only the nodes that start at 0 or all the nodes that don't have a predecessor? I would have thought the second as defined in the code:
```python
    @property
    def roots(self):
        if not hasattr(self, "_roots"):
            self._roots = set(self.nodes).difference(self.predecessor)
        return self._roots
```
and in that case the `if` here doesn't work

Reply:
The roots ideally should be the nodes without predecessor, however thats not the case in most tracking files, because some cells may be lost for ~50 or more timepoints and become untrackable. So usually as root I use self.time_nodes[0] I plan to change that at some point to self.roots.

Reply:
But the roots should have a big enough sublineage or at least more than one division.

Reply:
I dont remember, if we discussed this but roots should be a list, so that the order of the nodes will always be kept the same, and should also be modifiable.